src/coreclr/src/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -11,119 +11,153 @@
 ===========================================================*/
 
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Math
     {
+#pragma warning disable CA1401 // P/Invokes should not be visible
+
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall, CallingConvention = CallingConvention.Cdecl)]
         public static extern double Abs(double value);
 
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall, CallingConvention = CallingConvention.Cdecl)]
         public static extern float Abs(float value);
```

Comment:
Would it be simpler to have `Math.Abs(float)` forward to `MathF.Abs(float)`, that way we don't need to worry about also matching the types?

Reply:
I too think it would be cleaner and consistent to not have `float` related stuff in this file. But we are not matching the signature anymore with this QCall transition on the native side.

I will keep changes under src/libraries for a separate PR.

Reply:
It turned out I don't have a choice not to make that change in this PR. :)
There is no QCall macro which accepts the signature (like FCall has). I forgot that the `F` in `AbsFInternal` was helping to disambiguate the float signature. Tests were failing like:

```sh
    System.Tests.MathTests.Round_Float_Modes(x: -11.5, expected: -12, mode: AwayFromZero) [FAIL]
      Assert.Equal() Failure
      Expected: -12
      Actual:   -11
```

I have made that change in CoreCLR. Mono also has similar implementation, so I have only updated libraries using `#if CORECLR` in `Math.cs` and `MathF.cs`. I will update mono and cleanup the libraries code in a separate PR.